app/grandchallenge/algorithms/views.py

```diff
@@ -494,10 +496,35 @@ def create_upload(image_files):
             ci = interfaces[slug]
             if ci.is_image_kind:
                 if value:
-                    # create civ without image, image will be added when import completes
-                    civ = ComponentInterfaceValue.objects.create(interface=ci)
-                    civs.append(civ)
-                    upload_pks[civ.pk] = create_upload(value)
+                    widget = form.data[f"WidgetChoice-{ci.slug}"]
+                    if widget == WidgetChoices.IMAGE_SEARCH:
+                        try:
+                            (
+                                civ,
+                                created,
+                            ) = ComponentInterfaceValue.objects.get_or_create(
+                                interface=ci, image=value
+                            )
+                        except MultipleObjectsReturned:
```

Comment:
I don't think we should handle `MultipleObjectsReturned` - this _shouldn't_ happen, and if it does we should be made aware of it from sentry so that we can correct the entries in the db.